skrub/_drop_column_if_null.py

```diff
@@ -8,8 +8,23 @@
 
 
 class DropColumnIfNull(SingleColumnTransformer):
-    """Drop a single column if it contains only Null, NaN, or a mixture of null
-    values. If at least one non-null value is found, the column is kept."""
+    """Drop a single column if the fraction of Null or NaN values in the column
+    is larger than the given threshold.
+
+    By default, the threshold is set to `1.0`, so only columns that contain only
+    nulls or NaNs are dropped. Set the threshold to `None` to keep all columns.
+
+    Parameters
+    ----------
+    threshold : float, or None
+        Threshold of null values past which the column is dropped.
+    """
+
+    def __init__(self, threshold=1.0):
+        if threshold is not None:
+            assert 0.0 <= threshold <= 1.0, "Invalid value for the threshold."
```

Comment:
can we show the value of the threshold in the error message? also use a valueerror rather than an assertionerror. assertions would be used to stop execution if a bug in skrub code has led to a situation that shouldn't be possible. valueerror is how we communicate to the user that the value they provided is inappropriate

Reply:
in other words assertions are more of a debugging tool, not a way to validate user input. for example if the script is run with `python -O` or `PYTHONOPTIMIZE=1` they are not executed

Reply:
Noted, thanks for pointing that out. I changed it to a ValueError.